MapleServer2/PacketHandlers/Game/MasteryHandler.cs

```diff
@@ -81,8 +81,9 @@ public void HandleCraftItem(GameSession session, PacketReader packet)
                 return;
             }
 
-            // does the player have all the required ingredients for this recipe?
-            if (PlayerHasAllIngredients(session, recipe))
+            // does the player have all the required ingredients (and number of mesos) for this recipe?
+            if (PlayerHasAllIngredients(session, recipe)
+                && session.Player.Wallet.Meso.Amount >= recipe.GetMesosRequired())
```

Comment:
Why not make this check separately so you can alert the user that they don't have enough mesos?